sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -383,41 +426,40 @@ export class EventHubSender extends LinkEntity {
         this._context.connectionId,
         this.name
       );
-      const partitionKey = (options && options.partitionKey) || undefined;
-      const messages: AmqpMessage[] = [];
-      // Convert EventData to AmqpMessage.
-      for (let i = 0; i < events.length; i++) {
-        const message = toAmqpMessage(events[i], partitionKey);
-        message.body = this._context.dataTransformer.encode(events[i].body);
-        messages[i] = message;
-      }
-      // Encode every amqp message and then convert every encoded message to amqp data section
-      const batchMessage: AmqpMessage = {
-        body: message.data_sections(messages.map(message.encode))
-      };
-      // Set message_annotations, application_properties and properties of the first message as
-      // that of the envelope (batch message).
-      if (messages[0].message_annotations) {
-        batchMessage.message_annotations = messages[0].message_annotations;
-      }
-      if (messages[0].application_properties) {
-        batchMessage.application_properties = messages[0].application_properties;
-      }
-      for (const prop of messageProperties) {
-        if ((messages[0] as any)[prop]) {
-          (batchMessage as any)[prop] = (messages[0] as any)[prop];
+
+      let encodedBatchMessage: Buffer | undefined;
+      if (events instanceof EventDataBatch) {
+        encodedBatchMessage = events.batchMessage!;
```

Comment:
If I call `createBatch()` and then don't add any events to it. What is the expected behavior? `events.batchMessage` will be undefined then

Reply:
ohh yes, in that case we should add one more check in send() method something like:
```javascript
 if(eventData instanceof EventDataBatch && !eventData.batchMessage){
      const error = new TypeError(`Missing parameter "eventData", use tryAdd() function on the EventDataBatch to add events in a batch.`);
      log.error(`[${this._context.connectionId}] %O`, error);
      throw error;
    }
```

Reply:
Should we throw error or should we silently return?
cc @chradek?

Reply:
I think we should throw an error and let user know that they forgot to add events.

Reply:
Take a case where an event batch is created and passed to another function to add events. Once that function is done, the caller of the function then uses the send() method.
If the inner function couldn't fetch events, then we have the case of empty batch.

With the error model, the only way the outer function would know about empty message is to check `batch.batchMessage` which they may or may not have figured out how to do. Another option is to try/catch this error and ignore it.

Therefore, I feel that this is one case where we exiting the send() method without any error would be fine to do so. There is no downside to this model.

Reply:
I noticed that we currently throw an error in the `send` method if eventData is undefined. We're also unintentionally throwing an error if the eventData array passed in is empty (see #4307).

I'm personally in favor of treating an empty batch as a no-op, but whatever we decide I think we should do the same for the empty array case.

I'm also in favor of adding an `isEmpty` public getter to the batch if that is easier to discover and is helpful for a customer to make decisions with.

Reply:
We dont have to block the PR on this. But, lets discuss this with the others offline and get back to this in another PR